lib/gitlib.sh

```diff
@@ -201,3 +201,24 @@ gitlib::push_master () {
   logecho -n "Pushing$dryrun_flag master branch: "
   logrun -s git push$dryrun_flag origin master || return 1
 }
+
+##############################################################################
+# Ensure running with a synced repo.
+gitlib::repo_state () {
+  # Find remote
+  local remote=$(git -C $TOOL_ROOT remote -v |\
+                 awk '/kubernetes\/release \(fetch\)/ {print $1}')
+  local output=$(git -C $TOOL_ROOT fetch --dry-run $remote)
```

Comment:
maybe check if `git -C $TOOL_ROOT rev-list --right-only --count HEAD...$remote/master` is 0?

kind cargo-culting a lot, but I think this might do what we want.